test/utils/floating_point_utils.hpp

```diff
@@ -0,0 +1,69 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace fizzy::test
+{
+template <typename T, typename = std::enable_if<std::is_floating_point_v<T>>>
+struct FPUtils
+{
+    using Limits = std::numeric_limits<T>;
+
+    static constexpr auto NumMantissaBits = Limits::digits - 1;
+    static constexpr auto MantissaMask = (uint64_t{1} << NumMantissaBits) - 1;
+
+    static constexpr auto CanonicalNaNPayload = uint64_t{1} << (NumMantissaBits - 1);
+};
+
+template <typename T, typename = std::enable_if<std::is_floating_point_v<T>>>
+uint64_t as_uint64(T value) noexcept
+{
+    constexpr auto num_bytes = sizeof(T);
+    uint8_t bytes[num_bytes];
+    __builtin_memcpy(bytes, &value, num_bytes);
```

Comment:
Do we need to copy, cant' we just `reinterpret_cast<uint8_t*>(&value)`?

Reply:
That's actually undefined behavior, but I don't think compiler mind it. After revisiting, there should be simpler implementations ignoring endianness.